src/TextInput/TextInput.tsx

```diff
@@ -68,8 +70,22 @@ export class TextInput extends React.Component<Props, State> {
   static propTypes = propTypes;
   static defaultProps = defaultProps;
 
+  static validateProps(props) {
+    if (props.required && props.optional) {
+      throw new Error("You cannot pass both `required` and `optional` on a TextArea.");
```

Comment:
Let's also update this validation to `console.error`. Same with the analogous block in `TextArea`

Reply:
As I mentioned in a previous comment
>  Kept throw new error for 'required && optional' as that was there in textArea already and a new feature in textInput so no consumer is using it yet.

Any reason why console.warn should be preferred here

Reply:
Ah sorry about that! Totally missed that comment. Great point about there being no consumers of this API.

My main motivator here is consistency. I could imagine someone being confused by the fact that we throw an error in some circumstances and `console.warn` in others.

For validation in other components, we also already use `console.warn`, so that seems to be what we've converged on.

https://github.com/Clever/components/blob/3a9471a2c4ef72eebbadc17d160e24949a819fbf/src/Select/Select.tsx#L153-L165

Finally, given that nothing catastrophically breaks when you specify both `required` and `optional`, I don't think we need to throw a full error.